src/main/java/org/sunbird/workflow/service/impl/NotificationServiceImpl.java

```diff
@@ -179,23 +180,36 @@ public void sendNotification(WfRequest wfRequest) {
 							body = BP_MAIL_BODY.replace(STATE_NAME_TAG, wfStatusEntity.getCurrentStatus());
 							break;
 					}
-				}
-				else {
-					HashMap<String, Object> toValue = (HashMap<String, Object>) updatedFieldValue.get().get(TO_VALUE_CONST);
-					body = MAIL_BODY.replace(STATE_NAME_TAG, wfStatusEntity.getCurrentStatus())
-							.replace(FIELD_KEY_TAG, toValue.entrySet().iterator().next().getKey()).replace(TO_VALUE_TAG, (String) toValue.entrySet().iterator().next().getValue());
-					subjectLine = MAIL_SUBJECT.replace(STATE_NAME_TAG, wfStatusEntity.getCurrentStatus());
+				} else {
+					String fieldKey;
+					List<HashMap<String, Object>> values = wfRequest.getUpdateFieldValues();
+					for (Map<String, Object> fieldMap : values) {
+						if (!fieldMap.containsKey(Constants.TO_VALUE))
+							continue;
+						Map<String, Object> toValueMap = (Map<String, Object>) fieldMap.get(Constants.TO_VALUE);
+						fieldKey = toValueMap.entrySet().stream().findFirst().get().getKey();
+						if (Constants.APPROVE_STATE.equalsIgnoreCase(wfRequest.getAction()) && Constants.NAME.equalsIgnoreCase(fieldKey)) {
+							sendEmailnotification = false;
+						}
```

Comment:
disabling the notification in case of other states / when fieldKey is of different attributes -- is very bad thing to do.
Do we have any clarity on what are all use cases this code is used for? Did we test all the use cases?
So
if group / designation is approved - we shouldn't send notification?
if request is rejected -- we shouldn't send notification?